### ISSUE TITLE
Add option "checkIfExists" to Channel.fromPath method

### ISSUE THREAD

**Issue by ewels** [nextflow-io/nextflow#666]
It would be great if the `.fromPath()` channel generator could have a new option `checkIfExists` which, well, checks if it exists. Working with full file paths as well as globs.

See https://gitter.im/nextflow-io/nextflow?at=5ae0a290b01085166cbdb928 for discussion.

Phil

**Comment by ewels** [comment]
<img src="https://user-images.githubusercontent.com/465550/39259703-6cf89b74-48b7-11e8-9af8-97a86d9176d9.jpg" height="400">

**Comment by pditommaso** [comment]
SUPER-LOL 😂😂😂

**Comment by stevekm** [comment]
might also be able to do a custom `filter` step for this?

**Comment by pditommaso** [comment]
What do you mean for custom `filter` step?

**Comment by stevekm** [comment]
Like this:

```
Channel.from([
    ['bad1', file('input/bad1.vcf')],
    ['good1', file('input/good1.vcf')],
    ['foo', file('input/foo.vcf')]
    ])
    .filter { sampleID, sample_vcf ->
        def file_exists = sample_vcf.exists()
        if (! file_exists) println ">>> WARNING: File ${sample_vcf} does not exist and will not be included"
        file_exists
    }
    .set { demo_exists }
```

**Comment by pditommaso** [comment]
It looks this is off-topic.

**Comment by pditommaso** [comment]
Implementing this evil feature :) 

I was just thuinking the `checkIfExists` is a bit verbose, what about `shoudExist` or `mustExist`, something else ?

**Comment by pditommaso** [comment]
OK, I went for `checkIfExists` that's the most redable. Available for test with version `0.32.0u2-SNAPSHOT`.

**Comment by EasperOne** [comment]
lmao what did I just see
